### PR TITLE
Test 3 branch

### PR DESCRIPTION
**Pull request by mani1381**
1) Enable tests with blank R chunks.
2) add tests for Chapter 19 exercises.
3) Fixed minor spelling issues with Chapter 19 exercises, and chapter 20 of the textbook.

### CODE REVIEW

File: src/05-modelling-in-r/02-improving-models.Rmd [hunk 1]
```diff
@@ -1,616 +1,616 @@
...
+  #Create a note that indicates whether the concentration is < LOQ or < LOD
+  mutate(note = case_when(
+    instConc > LOQresid ~ "N/A",
+    instConc > LODresid ~ "<LOQ",
+    instConc < LODresid ~ "<LOD"))
+
+DT::datatable(FAESresultsfiltered)
+```
+
+Notice how it has taken the tap water values and sets all but one to 1/2 the LOQ - this is because we are not confident in the actual concentrations. This is artificially lowering their values, and since they're right on the edge of the LOQ in this case, it might be better to report them as-is instead of as 1/2 the LOQ, or to use the lower LOQ value we calculated earlier using the standard deviation of the intercept. If you were reporting these values in a publication, you would flag these values as < LOQ for clarity. As before, before reporting these values you would want to correct for the dilution factor and summarize them.
```
Comment (david-yz-liu) [inline src/05-modelling-in-r/02-improving-models.Rmd:572]: "set" is the correct verb tense here, so please revert that change

File: exercises/chapter 10/chapter10_tests.R [hunk 2]
```diff
@@ -13,11 +13,47 @@ run_all_chunks <- function(rmd, envir = globalenv()) {
   tempR <- file.path("temp.R")  # Create temporary file in the same directory
   on.exit(unlink(tempR))
   
-  purl(input = rmd, output = tempR)
+  purl(input = rmd, output = tempR)  # Extract R code from the Rmd file
   
-  # Source the temporary R script with chdir = TRUE so that the working directory is set to the Rmd file's directory
-  source(tempR, chdir = TRUE, local = envir)
```
Comment (david-yz-liu) [inline exercises/chapter 10/chapter10_tests.R:19]: @mani1381 can you please elaborate on the error this is solving?

**Review by david-yz-liu — COMMENTED** [review]
@mani1381 nice work. Don't worry about the larger diff, it's due to line ending differences (which isn't ideal, but doesn't impact your work). You can use the PR settings on GitHub to "hide whitespace" when showing the diff to see what your actual changes were.

File: exercises/chapter 10/chapter10_tests.R (hunk 2 above)
Comment (mani1381) [reply exercises/chapter 10/chapter10_tests.R:19]: Dear Professor Liu,

In the previous approach, after using the purl function, the entire R chunks in the Rmd file were converted into a single R script. By running this R script as a whole, an error in one part of the code, such as a syntax error, could cause issues in other parts. This meant that a single error could potentially fail all the tests.

In the new approach, I am utilizing the structure of the file after purl, where each chunk is separated by lines that start with "## ---------". By running each chunk separately, we can ensure that the tests remain independent of each other.

**Review by david-yz-liu — APPROVED** [review]

**Comment by david-yz-liu** [comment]
Great, thanks @mani1381!

**Pull request merged**